engine/src/main/java/org/hibernate/search/engine/mapper/schema/management/SchemaElementContext.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.mapper.schema.management;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Optional;
+
+import org.hibernate.search.util.common.annotation.Incubating;
+
+@Incubating
+public interface SchemaElementContext {
+	enum OutputFormat {
+		/**
+		 * Generate a JSON representation of this context.
```

Comment:
That's backend-specific though...  and it's not even useful at the moment since there's only one format.

I'd remove this completely. If this becomes necessary case we can expose additional `write` methods in `ElasticsearchSchemaElementContext`.